components/job-orchestration/job_orchestration/search_scheduler/search_scheduler.py

```diff
@@ -0,0 +1,306 @@
+        archives_for_search = get_archives_for_search(db_conn, search_config_obj)
+        if len(archives_for_search) == 0:
+            if set_job_status(db_conn, job_id, JobStatus.SUCCESS, JobStatus.PENDING):
+                logger.info(f"No matching archives, skipping job {job_id}.")
+            else:
+                logger.info(f"Marking job {job_id} with no archives to search cancelled.")
+                set_job_status(db_conn, job_id, JobStatus.CANCELLED)
+            continue
+
+        dispatch_search_job(archives_for_search, str(job_id), search_config_obj, results_cache_uri)
+        if set_job_status(db_conn, job_id, JobStatus.RUNNING, job_status):
+            logger.info(f"Dispatched job {job_id} with {len(archives_for_search)} archives to search.")
+        else:
+            logger.error(f"Marking job {job_id} with {len(archives_for_search)} archives to search cancelled.")
+            cancel_job(job_id)
+            set_job_status(db_conn, job_id, JobStatus.CANCELLED)
+
+
+def try_getting_results(result):
+    '''
+    https://github.com/celery/celery/issues/4084
+    Ideally we'd just do it this way, but because of the open bug listed above we have to use this
+    timeout based approach until we switch to redis result backend.
+    if not result.ready():
+        return None
+    return result.get()
+    '''
+    try:
+        return result.get(timeout=0.1)
+    except TimeoutError:
+        return None
+
+
+def check_job_status_and_update_db(db_conn):
+    global active_jobs
+
+    for job_id in list(active_jobs.keys()):
+        try:
+            returned_results = try_getting_results(active_jobs[job_id].async_task)
+        except Exception as e:
+            logger.error(f"Job `{job_id}` failed: {e}.")
+            # clean up
+            del active_jobs[job_id]
+            set_job_status(db_conn, job_id, JobStatus.FAILED)
+            continue
+
+        if returned_results is not None:
+            new_job_status = JobStatus.SUCCESS
+            for task_result in returned_results:
+                if not task_result['status']:
+                    task_id = task_result['task_id']
+                    new_job_status = JobStatus.FAILED
+                    logger.debug(f"Task {task_id} failed - result {task_result}.")
+
+            del active_jobs[job_id]
+
+            if set_job_status(db_conn, job_id, new_job_status, JobStatus.RUNNING):
+                if new_job_status != JobStatus.FAILED:
+                    logger.info(f"Completed job {job_id}.")
+                else:
+                    logger.info(f"Completed job {job_id} with failing tasks.")
+            else:
+                logger.info(f"Marking job {job_id} cancelled with no remaining archives to search.")
+                set_job_status(db_conn, job_id, JobStatus.CANCELLING)
+
+
+def handle_jobs(
+    db_conn,
+    results_cache_uri: str,
+    jobs_poll_delay: float,
+) -> None:
+    try:
```

Comment:
I would move this try into `main` since it's more of a final error check rather than a program error that this method should handle.